File: FbxImporter/ViewModels/FbxVertexData.cs
using System.Collections.Generic;

namespace FbxImporter.ViewModels;

// ReSharper disable once ClassNeverInstantiated.Global
// Is instantiated through Json deserialization.
public record FbxVertexData(float[] Position, float[] Normal, List<float[]> Tangents, List<float[]> UVs, string[] BoneNames, float[] BoneWeights);

File: FbxDataExtractor/FbxVertexData.cpp
#include "pch.h"
#include "FbxVertexData.h"

FbxDataExtractor::FbxVertexData::FbxVertexData()
{
	Position = gcnew array<float>(3);
	Normal = gcnew array<float>(3);
	Tangents = gcnew List<array<float>^>();
	UVs = gcnew List<array<float>^>();
	BoneNames = gcnew List<String^>();
	BoneWeights = gcnew List<float>();
}
